picard/metadata.py

```diff
@@ -244,6 +247,35 @@ def compare_to_release_parts(self, release, weights):
         except (ValueError, KeyError):
             pass
 
+        # Date Logic - in order of preference
+        # release has a date and it matches what our metadata had exactly.
+        # release has a date and it matches what our metadata had for year exactly.
+        # release has a date and it matches what our metadata had closely (year +/- 2).
+        # release has a date but we don't have one (all else equal, we prefer tracks that have non-blank date values)
+        # release has a no date (all else equal, we don't prefer this release since its date is missing)
+        # release has a date but it does not match ours(all else equal, its better to have an unknown date than a wrong date. since the unknown could actually be correct)
+        factor = 0.0
+        if "date" in release and release['date'] != '':
+            release_date = release['date']
+            release_year = extract_year_from_date(release_date)
+            if "date" in self:
+                metadata_date = self['date']
+                if release_date == metadata_date:
+                    factor = 1.00
+                else:
+                    metadata_year = extract_year_from_date(metadata_date)
+                    if release_year == metadata_year:
+                        factor = 0.95
```

Comment:
Brownie points if you make these constants with understandable names!

Reply:
done